smartsim/_core/_cli/build.py

```diff
@@ -331,84 +328,146 @@ def infer_torch_device(self) -> str:
             device = "gpu"
         return device
 
-    def install_torch(self, device: str = "cpu") -> None:
+    def check_py_torch_version(
+        self, device: str = "cpu", handle_conflict: bool = False
+    ) -> None:
         """Torch shared libraries installed by pip are used in the build
         for SmartSim backends so we download them here.
         """
-
-        if not self.build_env.check_installed("torch", self.versions.TORCH):
-            inferred_device = self.infer_torch_device()
-            if (inferred_device == "gpu") and (device == "cpu"):
-                logger.warning("CPU requested, but GPU backend is available")
-            _install_torch_from_pip(self.versions, device, self.verbose)
-        # if torch already installed, check the versions to make sure correct
-        # torch version is downloaded for that particular device
-        else:
-            installed = Version_(pkg_resources.get_distribution("torch").version)
-            if device == "gpu":
-                # if torch version is x.x.x+cpu
-                if "cpu" in installed.patch:
-                    msg = (
-                        "Torch CPU is currently installed but torch GPU requested. Uninstall all torch packages "
-                        + "and run the `smart build` command again to obtain Torch GPU libraries"
-                    )
-                    logger.warning(msg)
-
+        logger.info(f"Searching for a compatible TORCH install...")
+        if self.build_env.is_macos():
+            end_point = None
+            device_suffix = ""
+        else:  # linux
+            end_point = "https://download.pytorch.org/whl/torch_stable.html"
             if device == "cpu":
-                # if torch version if x.x.x then we need to install the cpu version
-                if "cpu" not in installed.patch and not self.build_env.is_macos():
-                    msg = (
-                        "Torch GPU installed in python environment but requested Torch CPU. "
-                        + " Run `pip uninstall torch torchvision` and run `smart build` again"
-                    )
-                    logger.error(msg)  # error because this is usually fatal
-            logger.info(f"Torch {self.versions.TORCH} installed in Python environment")
+                device_suffix = self.versions.TORCH_CPU_SUFFIX
+            elif device in ["gpu", "cuda"]:
```

Comment:
We don't recognize "cuda" as a device in SmartRedis, which may be confusing to users. Is this an artifact of adding support for different vendors' GPUs?  Suggest opening a ticket against SmartRedis to trigger a group conversation on this

Reply:
Good point! This was holdover code from the previous `_install_torch_from_pip` top level function that I stole. Not sure why `"cuda"` is treated as an acceptable value...

Removing and added a more strict `choices=['cpu', 'gpu']` param to our argparser